agent/xds/listeners.go

```diff
@@ -1662,3 +1667,66 @@ func makeCommonTLSContextFromFiles(caFile, certFile, keyFile string) *envoy_tls_
 
 	return &ctx
 }
+
+func validateListenerTLSConfig(tlsMinVersion types.TLSVersion, cipherSuites []types.TLSCipherSuite) error {
```

Comment:
Much of the rest of this file is just pushed around from the existing code used by `ingress-gateway`.